### PR TITLE
[NFC] update chunk manager API

### PR DESCRIPTION
**Pull request by feifeibear**
Changes some var names.
The biggest change is Chunk Manager.
append_tensor -> register_tensor.

### CODE REVIEW

File: colossalai/gemini/chunk/chunk.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def close_chunk(self):
         self.chunk_temp = None
 
         self.__scatter()
-        # always gathered chunk does not have shard
+        # always gathered chunk does not have shard chunk
```
Comment (1SAA) [inline colossalai/gemini/chunk/chunk.py:297]: why adding `chunk`?

Comment (feifeibear) [reply colossalai/gemini/chunk/chunk.py:297]: 这句话中文是什么？你写一下吧
gathered chunk是一个名词？always + 名词并不符合语法？

Comment (1SAA) [reply colossalai/gemini/chunk/chunk.py:297]: `consistently gathered chunk does not have shard attribute`

Comment (feifeibear) [reply colossalai/gemini/chunk/chunk.py]: DONE

**Pull request merged**